lib/src/main/java/com/diffplug/spotless/java/PalantirJavaFormatStep.java

```diff
@@ -0,0 +1,152 @@
+		return JVM_SUPPORT.getRecommendedFormatterVersion();
+	}
+
+	public static String defaultStyle() {
+		return DEFAULT_STYLE;
+	}
+
+	static final class State implements Serializable {
+		private static final long serialVersionUID = 1L;
+
+		/** The jar that contains the formatter. */
+		final JarState jarState;
+		final String stepName;
+		final String version;
+		final String style;
+
+		State(String stepName, String groupArtifact, String version, String style, Provisioner provisioner) throws Exception {
+			JVM_SUPPORT.assertFormatterSupported(version);
+			this.jarState = JarState.from(groupArtifact + ":" + version, provisioner);
+			this.stepName = stepName;
+			this.version = version;
+			this.style = style;
+		}
+
+		@SuppressWarnings({"unchecked", "rawtypes"})
+		FormatterFunc createFormat() throws Exception {
+			ClassLoader classLoader = jarState.getClassLoader();
+
+			// instantiate the formatter and get its format method
+			Class<?> optionsClass = classLoader.loadClass(OPTIONS_CLASS);
+			Class<?> optionsBuilderClass = classLoader.loadClass(OPTIONS_BUILDER_CLASS);
+			Method optionsBuilderMethod = optionsClass.getMethod(OPTIONS_BUILDER_METHOD);
+			Object optionsBuilder = optionsBuilderMethod.invoke(null);
+
+			Class<?> optionsStyleClass = classLoader.loadClass(OPTIONS_STYLE);
+			Object styleConstant = Enum.valueOf((Class<Enum>) optionsStyleClass, style);
+			Method optionsBuilderStyleMethod = optionsBuilderClass.getMethod(OPTIONS_BUILDER_STYLE_METHOD, optionsStyleClass);
+			optionsBuilderStyleMethod.invoke(optionsBuilder, styleConstant);
+
+			Method optionsBuilderBuildMethod = optionsBuilderClass.getMethod(OPTIONS_BUILDER_BUILD_METHOD);
+			Object options = optionsBuilderBuildMethod.invoke(optionsBuilder);
+
+			Class<?> formatterClazz = classLoader.loadClass(FORMATTER_CLASS);
+			Object formatter = formatterClazz.getMethod("createFormatter", optionsClass).invoke(null, options);
+			Method formatterMethod = formatterClazz.getMethod(FORMATTER_METHOD, String.class);
+
+			Function<String, String> removeUnused = constructRemoveUnusedFunction(classLoader);
+
+			Class<?> importOrdererClass = classLoader.loadClass(IMPORT_ORDERER_CLASS);
+			Method importOrdererMethod = importOrdererClass.getMethod(IMPORT_ORDERER_METHOD, String.class);
+
+			return JVM_SUPPORT.suggestLaterVersionOnError(version, (input -> {
+				String formatted = (String) formatterMethod.invoke(formatter, input);
+				String removedUnused = removeUnused.apply(formatted);
+				return (String) importOrdererMethod.invoke(null, removedUnused);
+			}));
+		}
+
+		private static Function<String, String> constructRemoveUnusedFunction(ClassLoader classLoader)
+				throws NoSuchMethodException, ClassNotFoundException {
+			Class<?> removeUnusedClass = classLoader.loadClass(REMOVE_UNUSED_CLASS);
+			Method removeUnusedMethod = removeUnusedClass.getMethod(REMOVE_UNUSED_METHOD, String.class);
+			return (x) -> (String) removeUnusedMethod.invoke(null, x);
+		}
```

Comment:
Cruft unless you have larger plans for how `removeUnusedImports` works. 

https://github.com/diffplug/spotless/blob/bb27596ed33d3029202c94940943926bba0e4229/lib/src/main/java/com/diffplug/spotless/java/RemoveUnusedImportsStep.java#L24-L36

Totally fine for you to build that into `palantirJavaFormat` itself, e.g. `palantirJavaFormat().removeUnusedImports(true)`, you can have the default be whatever you want, or non-configurable that it always removes, etc.

The existing setup is that users can pick either `eclipse()` or `googleJavaFormat()`, and then add `removeUnusedImports()` to that stack (which uses the latest available GJF under the hood). Perfectly reasonable for the `palantirJavaFormat` step to take a different approach.